### PR TITLE
Allow custom actor properties

### PR DESCRIPTION
**Pull request by gomezzz**
# Description

Summary of changes

* Added ability for custom properties
* Added automatic monitoring of them
* Added new test for custom properties
* Small refactoring, reformatting (did we want 100? 127? flake has 127 but that is actually a lot 🤔 )

## Resolved Issues

N/A

## How Has This Been Tested?

CI/CD, new test

## Related Pull Requests

N/A

### CODE REVIEW

File: paseos/tests/custom_property_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    # Add a custom property which tracks the total power consumption
+    prop_name = "total_power_consumption"
+
+    # Define the update function
+    def prop_update_fn(actor, dt, power_consumption):
+        return actor.get_custom_property(prop_name) + power_consumption * dt
+
+    ActorBuilder.add_custom_property(actor=sat1, property_name=prop_name, update_function=prop_update_fn, initial_value=0)
+    print(f"Actor custom props are now {sat1.custom_properties}")
```
Comment (GabrieleMeoni) [inline paseos/tests/custom_property_test.py:29]:
```suggestion
    print(f"Actor custom properties are now {sat1.custom_properties}")
```

**Review by GabrieleMeoni — CHANGES_REQUESTED** [review]
All good for me except one very minor request. I tested it, and it seems working.
About flake I would leave 127 chars, otherwise it becomes very annoying.

File: paseos/tests/custom_property_test.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    sim, sat1, earth = get_default_instance()
+    sat1 = ActorBuilder.get_actor_scaffold("sat1", SpacecraftActor, pk.epoch(0))
+    ActorBuilder.set_orbit(sat1, [10000000, 0, 0], [0, 8000.0, 0], pk.epoch(0), earth)
+    ActorBuilder.set_power_devices(sat1, 50000, 1000000, 1, paseos.PowerDeviceType.SolarPanel)
+
+    # Add a custom property which tracks the total power consumption
+    prop_name = "total_power_consumption"
+
+    # Define the update function
+    def prop_update_fn(actor, dt, power_consumption):
```
Comment (GabrieleMeoni) [inline paseos/tests/custom_property_test.py:25]: How do you specify "power_consumption"? This is not clear to me.

**Review by GabrieleMeoni — CHANGES_REQUESTED** [review]

File: paseos/tests/custom_property_test.py (hunk 2 above)
Comment (gomezzz) [reply paseos/tests/custom_property_test.py:25]: The same way as always. Either you run an activity with defined power consumption 

```py
# Register an activity that emulate event detection
sim.register_activity(
    "activity_A",
    activity_function=activity_function_A,
    power_consumption_in_watt=10
)

#Run the activity
sim.perform_activity("activity_A")
```

or you call 

```py
sim.advance_time(3600, power_consumption=10)
```

Nothing  changed. :)

**Review by GabrieleMeoni — APPROVED** [review]

**Pull request merged**